library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt

```diff
@@ -46,13 +29,34 @@ public class ChuckerInterceptor private constructor(
      */
     public constructor(context: Context) : this(Builder(context))
 
-    private val context = builder.context
-    private val collector = builder.collector ?: ChuckerCollector(context)
-    private val maxContentLength = builder.maxContentLength
-    private val cacheDirectoryProvider = builder.cacheDirectoryProvider ?: CacheDirectoryProvider { context.filesDir }
-    private val alwaysReadResponseBody = builder.alwaysReadResponseBody
     private val headersToRedact = builder.headersToRedact.toMutableSet()
-    private val requestProcessor = RequestProcessor(context, collector, maxContentLength, headersToRedact)
+
+    private val collector: ChuckerCollector
+
+    private val requestProcessor: RequestProcessor
+
+    private val responseProcessor: ResponseProcessor
+
+    init {
+        val context = builder.context
+        val maxContentLength = builder.maxContentLength
```

Comment:
Is there a reason why you moved those initialisation inside the `init{}` block?

Reply:
They aren't used by anything else except for processors. There is no need to keep them in the interceptor.

Reply:
I guess it could be moved out of init block and reuse the values from the builder.